x/programs/runtime/map.go

```diff
@@ -118,28 +87,19 @@ func (m *MapModule) getBytesFn(ctx context.Context, mod api.Module, id int64, ke
 		m.log.Error("key or value length is negative")
 		return mapErr
 	}
-	_, ok := GlobalStorage.state[id]
-	if !ok {
-		m.log.Error("failed to find program id in storage")
-		return mapErr
-	}
 	buf, ok := utils.GetBuffer(mod, keyPtr, uint32(keyLength))
 	if !ok {
 		return mapErr
 	}
-	val, ok := GlobalStorage.state[id][string(buf)]
-	if !ok {
-		return mapErr
-	}
 
-	result, err := mod.ExportedFunction("alloc").Call(ctx, uint64(valLength))
+	// Get the value from the database
+	val, err := GetValue(m.db, uint64(id), buf)
 	if err != nil {
-		m.log.Error("failed to allocate memory for value: %v", zap.Error(err))
 		return mapErr
 	}
-	ptr := result[0]
-	// write to memory
-	if !mod.Memory().Write(uint32(ptr), val) {
+
+	ptr, err := utils.WriteBuffer(ctx, mod, val)
```

Comment:
we need to dealloc here right?

Reply:
no, the guest takes control of the memory so rust will free it.